packages/ui/src/providers/Config/createClientConfig/getCreateMappedComponent.tsx

```diff
@@ -27,10 +27,17 @@ export function getCreateMappedComponent({
   const createSingleMappedComponent = (
     payloadComponent: { ReactComponent: React.FC<any> } | PayloadComponent,
     key: number | string,
-    props: JsonObject,
+    props: {
+      serverProps: JsonObject
+    } & JsonObject,
     Fallback: React.FC<any>,
     identifier: string,
   ): MappedComponent => {
+    const {
+      serverProps: componentServerProps, // must keep this in place to ensure it is omitted from the spread
```

Comment:
In the `CreateMappedComponent` type (different file), can we type the props argument the same way? So
```
{
      serverProps: JsonObject
    } & JsonObject 
```

instead of just `object`.

+ we could change `serverProps` from `JsonObject` to just `object`, since server props don't have to be serializable. Right now, JsonObject is typed weakly, but I'd love to change that in the future.

Also, what do you think about typing it like this instead:

```
{
  serverProps: object
  clientProps: JsonObject
}
```

That way, it would be consistent with our `PayloadComponent` type, and we'd need one less spread here.

Reply:
I think this is a good call, it will make server props much more explicitly separate from client props so don't have to rely on the `...rest` destructuring pattern.